docs/service/webapp.rst

```diff
@@ -153,46 +132,45 @@ Here's an example in a template::
   {% endif %}
 
 
-When you add a new permission here they will automatically appear on the `Your
-Permissions <https://crash-stats.mozilla.com/permissions/>`_ page.
-
-
-Troubleshooting
-===============
-
-If you have set up your webapp but you can't sign in, it could very well be
-because some configuration is wrong compared to how you're running the webapp.
+When you add a new permission via ``signal.py``, it will automatically appear
+on the `Your Permissions <https://crash-stats.mozilla.com/permissions/>`_ page
+for the users in that group.
 
-If this is the problem go to ``http://localhost:8000/_debug_login``.
 
-This works for both production and development. If you're running in production
-you might not be using ``localhost:8000`` so all you need to remember is to go
-to ``/_debug_login`` on whichever domain you will use in production.
+Static Assets
+=============
+In the development environment, the ``STATIC_ROOT`` is set to
+``/tmp/crashstats-static/`` rather than ``/app/webapp-django/static``.
+The process in the container creates files with the uid 10001, and Linux users
+will have permissions-related problems if these are mounted on the host
+computer.
 
-If web services are not starting up, ``/var/log/nginx/`` is a good place to
-look.
+The problem this creates is that ``/tmp/crashstats-static/`` is ephemeral
+and any changes there disappear when you stop the container.
 
-If you are not able to log in to the crash-stats UI, try hitting
-``http://crash-stats/_debug_login``
+If you are on macOS or Windows, then Docker uses a shared file system that
+creates files with your user ID. This makes it safe to persist static assets,
+at the cost of slower file system performance.
 
-If you are having problems with crontabber jobs, this page shows you the
-state of the dependencies: ``http://crash-stats/crontabber-state/``
+If you want static assets to persist between container restarts, then you
+can override ``STATIC_ROOT`` in ``my.env`` to return it to the ``app`` folder::
 
-If you're seeing "Internal Server Error", you can get Django to send you email
-with stack traces by adding this to
-``/data/socorro/webapp-django/crashstats/settings/base.py``:
+    STATIC_ROOT=/app/static
 
-::
+Alternatively, you can mount ``/tmp/crashstats-static/`` using ``volumes``
+in a ``docker-compose.override.yml`` file:
 
-  # Recipients of traceback emails and other notifications.
-  ADMINS = (
-      ('Your Name', 'your_email@domain.com'),
-  )
-  MANAGERS = ADMINS
+.. code-block:: yaml
 
+    version: "2"
+    services:
+      webapp:
+        volumes:
+          # Persist the static files folder
+          - ./static:/tmp/crashstats-static
```

Comment:
For the record, this is my fault. It's terrible. I have some thoughts on how to fix it and make it less terrible, but I haven't done it, yet.

Reply:
I feel this is a limitation of Docker environments, and this isn't a bad solution, especially since you use long-running containers in your personal workflow. I'm happy to document now and improve later.

I've also struggled with this on Kuma. Our solution was [to set ``UID``](https://github.com/mozilla/kuma/blob/8ff9b279ce4b0c4f446b5c8591afdea16f43af66/.env-dist.dev#L3-L8), so the in-container user matched the host user. I'm not thrilled with that approach either, but it does make Linux work more like macOS.  For production, we have a derived image that explicitly gathers assets.

I've used the in-container directory trick as well, for installing npm packages and avoiding s system mismatch on ``node_modules``.

Reply:
We allow people to set the uid/gid to use in `my.env`. That covers the permissions issues that differ between Linux and OSX. That's what step 3 here is about:

https://socorro.readthedocs.io/en/latest/localdevenvironment.html#setup-quickstart

If I recall correctly, the problem here is that the static assets get built in the image, but then in `docker-compose`, we stomp on `/app` with the current directory, so then the static assets that got built during `make build` disappear. So then you have two groups of static assets and it's hard to know which one you're adjusting. Or something along those lines.